application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,269 @@
+package org.togetherjava.tjbot.commands.bytecode;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.events.message.MessageDeleteEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.MessageUpdateEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.commands.MessageReceiver;
+import org.togetherjava.tjbot.imc.CompilationResult;
+import org.togetherjava.tjbot.imc.CompileInfo;
+import org.togetherjava.tjbot.imc.InMemoryCompiler;
+import org.togetherjava.tjbot.imc.JavacOption;
+import org.togetherjava.tjbot.javap.Javap;
+import org.togetherjava.tjbot.javap.JavapOption;
+
+import javax.tools.Diagnostic;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Bytecode command that uses the message context commands Example usage:
+ * <p>
+ *
+ * <pre>
+ * {@code
+ *     ```java
+ *     class Test {
+ *         public void foo() {
+ *             System.out.println("Hello World!");
+ *         }
+ *     }
+ *     ```
+ * }
+ * </pre>
+ *
+ * then apply the application
+ */
+public final class BytecodeCommand extends BotCommandAdapter
+        implements MessageContextCommand, MessageReceiver {
+    private static final String CODE_BLOCK_OPENING = "```\n";
+    private static final String CODE_BLOCK_CLOSING = "\n```";
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
+    private final Map<Long, List<Long>> userToBotMessages = new HashMap<>();
+
+    public BytecodeCommand() {
+        super(Commands.message("View bytecode"), CommandVisibility.GUILD);
+    }
+
+    // Fresh compile when a user uses the message context command on one of their messages
```

Comment:
is this supposed to be javadoc or a comment? if comment, put it inside the method. otherwise it might not move with the method when its refactored.

Reply:
you originally told me to add that. I'll just remove it since imo it's totally pointless

Reply:
im not against it. it seems valuable. im just telling u to put the comment **inside** the method. not above it. otherwise it, a year later, it will not sit next to the method anymore but somewhere else lonely in the file.